packages/components/src/internal/components/base/Grid.tsx

```diff
@@ -273,15 +284,18 @@ class GridBody extends PureComponent<GridBodyProps> {
                     'grid-row': r % 2 === 1,
                 })}
             >
-                {columns.map((column: GridColumn, c: number) =>
-                    column.tableCell ? (
+                {columns.map((column: GridColumn, c: number) => {
+                    const columnMeta = columnMetaData?.[c];
+                    const hasWidthOverride = !!columnMeta?.width || !!columnMeta?.minWidth;
+                    const className = hasWidthOverride && !column.tableCell ? 'grid-col-with-width' : undefined;
+                    return column.tableCell ? (
```

Comment:
It seems to me that this is an exact use case for the `tableCell` override. I would much prefer that we don't introduce column metadata / additional complexity to the base grid rendering if we can avoid it.

Reply:
tableCell would work for grid body cells, but grid header would still need some additional config/meta to have the correct width. I could revert the change for GridBody and have EditableGrid inputCellFactory determine the body cell instead. However, I'm not sure that's a better/cleaner approach since GridHeader still needs width meta passed in for th. So a user of Grid ended up having to pass in columnMeta for header, but then also have to provide their cell implementation for the body.